src/main/java/com/tabnine/Initializer.java

```diff
@@ -65,6 +72,26 @@ private void initialize() {
         CapabilitiesService.getInstance().init();
         TabnineUpdater.pollUpdates();
         PluginInstaller.addStateListener(instanceOfUninstallListener());
+      } else if (Config.IS_ON_PREM && !DIALOG_SHOWED.get()) {
+        DIALOG_SHOWED.set(true);
+        Optional<String> cloud2Url = StaticConfig.getTabnineEnterpriseHost();
+        if (cloud2Url.isPresent()) {
+          Logger.getInstance(getClass()).info(String.format("Tabnine Enterprise host is configured: %s", cloud2Url.get()));
+          SetCustomRepository(cloud2Url.get());
+        } else {
```

Comment:
the `else` clause here is not right - you can get here by having `DIALOG_SHOWED == true` and `Config.IS_ON_PREM == false` - i.e., the else here doesn't necessarily mean that you're onprem.

Reply:
that an `else` for `if (cloud2Url.isPresent())`, not the one in the upper scope